bootstrap.js

```diff
@@ -7,54 +7,61 @@ const index = fs.readFileSync('index.html', 'utf8');
 const ResponseBuilder = require('./app/ResponseBuilder');
 
 module.exports = app => {
-    app.get('/*', (req, res) => {
+    app.all('/*', (req, res) => {
         const responseBuilder = new ResponseBuilder(res);
-        
+
         const requestedUrl = req.url.slice(1);
         const corsBaseUrl = '//' + req.get('host');
-        
-        console.info(req.protocol + '://' + req.get('host') + req.url);
-        
+
+        console.info(req.method + ' ' + req.protocol + '://' + req.get('host') + req.url);
+
         if(requestedUrl == ''){
             res.send(index);
             return;
         }
 
+        let headers = Object.assign({}, req.headers);
+
+        // This prevents "Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames: Host: localhost"
+        delete headers.host;
+
+        // This prevents "Received response with content-encoding: gzip, but failed to decode it"
+        delete headers['accept-encoding'];
+
         request({
+            method: req.method,
             uri: requestedUrl,
             resolveWithFullResponse: true,
-            headers: {
-                'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.116 Safari/537.36'
```

Comment:
Could we preserve this UA please?

Reply:
If the target system depends on the UA being correct, overwriting it will just break things. To me this behavior was really unexpected.